lib/core-integration/src/Test/Integration/Scenario/API/Shelley/StakePools.hs

```diff
@@ -524,6 +527,26 @@ spec = describe "SHELLEY_STAKE_POOLS" $ do
             , expectErrorMessage errMsg403NotDelegating
             ]
 
+    it "STAKE_POOLS_QUIT_03 - Can quit with rewards"
+        $ \ctx -> runResourceT $ do
+        (w, _) <- rewardWallet ctx
+
+        pool:_:_ <- map (view #id) . snd
+            <$> unsafeRequest @[ApiStakePool]
+                ctx (Link.listStakePools arbitraryStake) Empty
+        joinStakePool @n ctx pool (w, fixturePassphrase) >>= flip verify
+            [ expectResponseCode HTTP.status202
+            , expectField #depositTaken (`shouldBe` (Quantity 0))
```

Comment:
hmm, shouldn't this be `Quantity depositAmt` ?

Reply:
The stake key is already registered because this is a `rewardWallet`. Added a comment to the other test, but can add one here as well.

Reply:
aha, ok. so this is shelley wallet with gathered some rewards :+1:  (---> `rewardWallet` rather than `fixtureWallet`). And we do it in order not to wait. Then here we are really rejoining.

Reply:
ok, I was expecting the flow more like in suite `STAKE_POOLS_JOIN_04 - Rewards accumulate`

BTW. you need to correct there : 
```
652         -- Can't quite if unspoiled rewards.                                                                                                                                                                   
653         quitStakePool @n ctx (w, fixturePassphrase) >>= flip verify                                                                                                                                            
654             [ expectResponseCode HTTP.status403                                                                                                                                                                
655             , expectErrorMessage errMsg403NonNullReward                                                                                                                                                        
656             ]    
```
No anymore valid after your change.